metric_learn/base_metric.py

```diff
@@ -107,7 +179,7 @@ def score(self, pairs, y):
     return roc_auc_score(y, self.decision_function(pairs))
 
 
-class _QuadrupletsClassifierMixin:
+class _QuadrupletsClassifierMixin(BaseMetricLearner):
 
   def predict(self, quadruplets):
```

Comment:
it would be more logical if `predict` would compute accuracy on quadruplets (proportion of quadruplets correctly ordered) and `score` would compute the difference between distances

Reply:
I agree that having a `predict` and a `decision_function` doing the same thing is not ideal, and in the case of quadruplets this can be fixed easily since there is no threshold to fix like in pairs
But I did not really get what you mean ? 
Since `score` is the scikit-learn like function for scoring it should return one scalar, whereas `predict` should return a sample-wise output
I guess the most coherent with scikit-learn would be that `predict` would output a binary sign (or 0 or 1) depending on the ordering of pairs in the quadruplet, and `decision_function` would return the differences between distances, since it is a float-type score (like `predict_proba` but without being necessarily between 0 and 1)
`score` would return the accuracy of the predict function (proportion of quadruplets correctly ordered)

Reply:
Yes, this is what I meant, sorry for the confusion